python/paddle/fluid/tests/unittests/test_cuda_graph.py

```diff
@@ -236,6 +236,16 @@ def __getitem__(self, idx):
             self.assertTrue(np.array_equal(actual_x, x.numpy()))
             self.assertTrue(np.array_equal(actual_y, y.numpy()))
 
+    def test_dev_ctx_alloc(self):
+        if not can_use_cuda_graph():
+            return
+
+        x = paddle.to_tensor([2], dtype='float32')
+        graph = CUDAGraph()
+        graph.capture_begin()
+        y = paddle.cast(x, dtype='float16')
+        graph.capture_end()
```

Comment:
Add more complex cases?

Reply:
Very few op use dev_ctx.Alloc, cast is one of them. So just call a single cast here.